### PR TITLE
Development

### PR DESCRIPTION
**Pull request by BrokenJam**
Please review my code

### CODE REVIEW

File: CMP1903_A1_2324.csproj [hunk 1]
```diff
@@ -0,0 +1,10 @@
+﻿<Project Sdk="Microsoft.NET.Sdk">
+
+  <PropertyGroup>
+    <OutputType>Exe</OutputType>
+    <TargetFramework>net8.0</TargetFramework>
+    <ImplicitUsings>enable</ImplicitUsings>
+    <Nullable>enable</Nullable>
+  </PropertyGroup>
+
+</Project>
```
[the remark this answers is not in the document]
Comment (KacperUoL) [reply CMP1903_A1_2324.csproj:10]: Use of internal variables may not be neccessary

File: CMP1903_A1_2324/Game.cs [hunk 2]
```diff
@@ -15,7 +15,25 @@ internal class Game
...
 
-        //Methods
+        internal Die[] Dice = new Die[3];
+        internal int[] diceTotal = new int[3];
+        internal int sumOfDie;
+        internal int currentRoll;
 
+        //Methods
+        // Kacper - All criterias are met and the code works as intended and also follows the styling guidelines, code is relatively easy to understand and read 
+        public int[] GameMethod()
```
Comment (KacperUoL) [inline CMP1903_A1_2324/Game.cs:25]: Method does what is required and is written in the write style :)

File: CMP1903_A1_2324/Testing.cs [hunk 3]
```diff
@@ -4,17 +4,25 @@
...
         //Method
+        internal void TestingMethod() //Kacper - Creating a new method that is used to loop through the rolls in order to anaylse is good and follows the guidelines and what is being asked
+        {
+            Game GameObject = new Game();
+            int[] diceTotal = new int[3];
+            diceTotal = GameObject.GameMethod();
+
+            for (int i = 0; i < 3; i++)
+            {
+                Debug.Assert(diceTotal[i] >= 1 && diceTotal[i] <= 6, "Rolls within range"); // Kacper - Testing is successfull and outputs the correct messages when the condtions are met
```
Comment (KacperUoL) [inline CMP1903_A1_2324/Testing.cs:22]: In built function used as stated in brief, used correctly and output errors if criteria not met

File: CMP1903_A1_2324/Program.cs [hunk 4]
```diff
@@ -14,6 +15,14 @@ static void Main(string[] args)
              * Create a Game object and call its methods.
              * Create a Testing object to verify the output and operation of the other classes.
              */
+
+            // Kacper - appears to all be working and is written to the brief and what is asked to be done 
+            // Kacper - Code also follows the style guidelines 
+
```
Comment (KacperUoL) [inline CMP1903_A1_2324/Program.cs:21]: Object created for both testing and game follows the brief and styling guide

File: CMP1903_A1_2324/Program.cs [hunk 5]
```diff
@@ -1,6 +1,7 @@
 ﻿using System;
 using System.Collections.Generic;
 using System.Linq;
+using System.Security.Cryptography.X509Certificates;
 using System.Text;
 using System.Threading.Tasks;
```
Comment (KacperUoL) [inline CMP1903_A1_2324/Program.cs:7]: Could comment what each function does briefly to make it easier for user to read

**Review by KacperUoL — COMMENTED** [review]
good code

File: CMP1903_A1_2324/Game.cs
Comment (finob2805) [inline CMP1903_A1_2324/Game.cs:1]: Code does what is outlined and required.
Easy to read and understand
Good use of array

File: CMP1903_A1_2324/Program.cs
Comment (finob2805) [inline CMP1903_A1_2324/Program.cs:1]: All objects are created as required
All methods are called
Easy to understand

File: CMP1903_A1_2324/Testing.cs
Comment (finob2805) [inline CMP1903_A1_2324/Testing.cs:1]: Functionality is implemented well
Debug.Assert() is used however I believe that the error message should be Total and Rolls NOT within range as that part only activates when the condition is false
Easy to read

**Review by finob2805 — COMMENTED** [review]
Overall the code is efficient and does what the brief requires.
Variable names could be more in line with the C# guideline
Comments could be utilised to further explain what each part of the code does
Errors are handled however error message needs to be changed I believe

**Pull request merged**